pynumdiff/utils/utility.py

```diff
@@ -191,43 +127,18 @@ def peakdet(v, delta, x=None):
     return np.array(maxtab), np.array(mintab)
 
 
-# simple finite difference
-def finite_difference(x, dt):
-    """
-    :param x: (np.array of floats, 1xN) time series to differentiate
-    :param dt: (float) time step
-    :param params: (list): (int, optional) number of iterations ignored if 'iterate' not in options
-    :param options: (bool) iterate the finite difference method (smooths the estimates)
-    :return: x_hat: smoothed x; dxdt_hat: derivative of x
-    """
-    # Calculate the finite difference
-    dxdt_hat = np.diff(x)/dt
-    # Pad the data
-    dxdt_hat = np.hstack((dxdt_hat[0], dxdt_hat, dxdt_hat[-1]))
-    # Re-finite dxdt_hat using linear interpolation
-    dxdt_hat = np.mean((dxdt_hat[0:-1], dxdt_hat[1:]), axis=0)
-
-    return x, dxdt_hat
-
-
 # Trapazoidal integration, with interpolated final point so that the lengths match.
 def integrate_dxdt_hat(dxdt_hat, dt):
-    """
-    Wrapper for scipy.integrate.cumulative_trapezoid to integrate dxdt_hat that ensures the integral has the same length
-
-    :param dxdt_hat: estimate derivative of timeseries
-    :type dxdt_hat: np.array
+    """Wrapper for scipy.integrate.cumulative_trapezoid to integrate dxdt_hat that ensures the integral has the same length
 
-    :param dt: time step in seconds
-    :type dt: float
+    :param np.array[float] dxdt_hat: estimate derivative of timeseries
+    :param float dt: time step in seconds
 
-    :return: integral of dxdt_hat
-    :rtype: np.array
+    :return: **x_hat** (np.array[float]) -- integral of dxdt_hat
     """
     x = scipy.integrate.cumulative_trapezoid(dxdt_hat)
-    first_value = x[0] - np.mean(dxdt_hat[0:1])
```

Comment:
What is this doing? You have an array `dxdt_hat`, and you take the slice `[0:1]`, which *includes* the 0 index and *excludes* the 1 index. So this is equivalent to just taking `dxdt_hat[0]`. Then you take the mean of that single entry, which returns that same scalar value again, and then you subtract it off `x[0]`. But `dxdt[0]` and `x[0]` have different units, right? I'm confused what this line is doing.